tests/raftstore/test_split_region.rs

```diff
@@ -249,3 +253,103 @@ fn test_server_delay_split_region() {
     let mut cluster = new_server_cluster(0, 3);
     test_delay_split_region(&mut cluster);
 }
+
+struct DropSnapshotFailure {
+    count: AtomicUsize,
+}
+
+impl Filter for DropSnapshotFailure {
+    fn before(&self, m: &RaftMessage) -> bool {
+        if m.get_message().get_msg_type() != raftpb::MessageType::MsgSnapStatus {
+            return false;
+        }
+
+        // ignore the first snapshot failure message
+        self.count.fetch_add(1, Ordering::SeqCst) < 1
+    }
+}
+
+fn test_split_overlap_snapshot<T: Simulator>(cluster: &mut Cluster<T>,
+                                             drop_snapshot_failure: bool) {
+    // disable log GC.
+    cluster.cfg.store_cfg.raft_log_gc_threshold = 300000;
+
+    // util::init_log();
+
+    // We use three nodes 1, 2, 3 for this test.
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    // guarantee 1 is leader
+    cluster.transfer_leader(1, util::new_peer(1, 1));
+    cluster.reset_leader_of_region(1);
```

Comment:
yes, after calling transfer_leader, the origin leader map may still cache old leader, we must reset it explicitly. 
of course, we can call other must_put to do it, but here reset is ok too.